core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2067,8 +2067,8 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
-   * Run a job on all partitions in an RDD and pass the results to a handler function, but take
-   * a function of type `Iterator[T] => U` instead of `(TaskContext, Iterator[T]) => U`.
+   * Run a job on all partitions in an RDD and pass the results to a handler function, this time the
```

Comment:
I wasn't specific enough here ... I don't think this is the best way to write it. Both functions do X. One lets you do X with an additional argument. Both should say they do X. The one with the additional argument should mention its additional argument. Saying "the other one doesn't have this arg" seems reversed. I'm not that against merging it but because the point is to improve docs we should probably take extra care to make the wording clear.